components/ActivityCard.js

```diff
@@ -1,34 +1,46 @@
 import Image from "next/image";
+import Link from "next/link";
 import styled from "styled-components";
+import { useRouter } from "next/router";
 
-export default function ActivityCard({ title, categories, imageUrl }) {
-  return (
-    <StyledCardSection>
-      <StyledImageDiv>
+export default function ActivityCard({ id, title, categories, imageUrl }) {
+  const router = useRouter();
+
+  const handleClick = () => {
```

Comment:
Please remove your handleClick function as you're not using it anymore (you are now using Link)